CONTRIBUTING.md

```diff
@@ -52,6 +52,12 @@ and run binderhub on the host system.
 
 All features should work, including building and launching.
 
+9. Running unit tests
+
+  ```bash
+  python setup.py test
```

Comment:
Is there an advantage to running it like this over just `pytest`?

Reply:
This way, it should install anything under `test_dependencies` before running the tests. However, it could easily be substituted for a combination of `pip install pytest; py.test -v tests`

Reply:
Won’t using test_requires cause the dependencies to be installed as eggs dumped in the current dir, instead of regular package installation in your virtualenv?  IMO it’s best to have a test-requirements file and use tox to abstract virtualenv creation, deps installation and test runner options.

Reply:
yes, test_requires and setup_requires can't be relied upon thanks to setuptools. A simple `test-requirements.txt` would be simpler and more reliable.

I strongly prefer to *avoid* the overhead and indirection of of tox, though.

Reply:
Well the good thing with tox is that it builds on top of your existing settings and toolings.  In my projects I run commands such as flake8 or pytest directly most of the time (and they find their options in tox.ini), and only to make sure things are right in a clean isolated environment I call tox.